File: dmn-test-cases/standard/tck/1.1/cl3/0031-static-user-defined-functions/translator/expected/java/dmn/type/TFnInvocationNamedResult.java
package type;

import java.util.*;

@javax.annotation.Generated(value = {"itemDefinitionInterface.ftl", "tFnInvocationNamedResult"})
@com.fasterxml.jackson.annotation.JsonPropertyOrder(alphabetic = true)
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(as = type.TFnInvocationNamedResultImpl.class)
public interface TFnInvocationNamedResult extends com.gs.dmn.runtime.DMNType {
    static TFnInvocationNamedResult toTFnInvocationNamedResult(Object other) {
        if (other == null) {
            return null;
        } else if (TFnInvocationNamedResult.class.isAssignableFrom(other.getClass())) {
            return (TFnInvocationNamedResult)other;
        } else if (other instanceof com.gs.dmn.runtime.Context) {
            TFnInvocationNamedResultImpl result_ = new TFnInvocationNamedResultImpl();
            if (((com.gs.dmn.runtime.Context)other).keySet().contains("subResult")) {
                result_.setSubResult((java.lang.Number)((com.gs.dmn.runtime.Context)other).get("subResult"));
            } else {
                return  null;
            }
            if (((com.gs.dmn.runtime.Context)other).keySet().contains("subResultMixed")) {
                result_.setSubResultMixed((java.lang.Number)((com.gs.dmn.runtime.Context)other).get("subResultMixed"));
            } else {
                return  null;
            }
            if (((com.gs.dmn.runtime.Context)other).keySet().contains("divisionResultNamed")) {
                result_.setDivisionResultNamed((java.lang.Number)((com.gs.dmn.runtime.Context)other).get("divisionResultNamed"));
            } else {
                return  null;
            }
            if (((com.gs.dmn.runtime.Context)other).keySet().contains("multiplicationResultNamed")) {
                result_.setMultiplicationResultNamed((java.lang.Number)((com.gs.dmn.runtime.Context)other).get("multiplicationResultNamed"));
            } else {
                return  null;
            }
            return result_;
        } else if (other instanceof com.gs.dmn.runtime.DMNType) {
            return toTFnInvocationNamedResult(((com.gs.dmn.runtime.DMNType)other).toContext());
        } else {
            throw new com.gs.dmn.runtime.DMNRuntimeException(String.format("Cannot convert '%s' to '%s'", other.getClass().getSimpleName(), TFnInvocationNamedResult.class.getSimpleName()));
        }
    }

    @com.fasterxml.jackson.annotation.JsonGetter("subResult")
    java.lang.Number getSubResult();

    @com.fasterxml.jackson.annotation.JsonGetter("subResultMixed")
    java.lang.Number getSubResultMixed();

    @com.fasterxml.jackson.annotation.JsonGetter("divisionResultNamed")
    java.lang.Number getDivisionResultNamed();

    @com.fasterxml.jackson.annotation.JsonGetter("multiplicationResultNamed")
    java.lang.Number getMultiplicationResultNamed();

    default com.gs.dmn.runtime.Context toContext() {
        com.gs.dmn.runtime.Context context = new com.gs.dmn.runtime.Context();
        context.put("subResult", getSubResult());
        context.put("subResultMixed", getSubResultMixed());
        context.put("divisionResultNamed", getDivisionResultNamed());
        context.put("multiplicationResultNamed", getMultiplicationResultNamed());
        return context;
    }

    default boolean equalTo(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        TFnInvocationNamedResult other = (TFnInvocationNamedResult) o;
        if (this.getDivisionResultNamed() != null ? !this.getDivisionResultNamed().equals(other.getDivisionResultNamed()) : other.getDivisionResultNamed() != null) return false;
        if (this.getMultiplicationResultNamed() != null ? !this.getMultiplicationResultNamed().equals(other.getMultiplicationResultNamed()) : other.getMultiplicationResultNamed() != null) return false;
        if (this.getSubResult() != null ? !this.getSubResult().equals(other.getSubResult()) : other.getSubResult() != null) return false;
        if (this.getSubResultMixed() != null ? !this.getSubResultMixed().equals(other.getSubResultMixed()) : other.getSubResultMixed() != null) return false;

        return true;
    }

    default int hash() {
        int result = 0;
        result = 31 * result + (this.getDivisionResultNamed() != null ? this.getDivisionResultNamed().hashCode() : 0);
        result = 31 * result + (this.getMultiplicationResultNamed() != null ? this.getMultiplicationResultNamed().hashCode() : 0);
        result = 31 * result + (this.getSubResult() != null ? this.getSubResult().hashCode() : 0);
        result = 31 * result + (this.getSubResultMixed() != null ? this.getSubResultMixed().hashCode() : 0);
        return result;
    }

    default String asString() {
        StringBuilder result_ = new StringBuilder("{");
        result_.append("divisionResultNamed=" + getDivisionResultNamed());
        result_.append(", multiplicationResultNamed=" + getMultiplicationResultNamed());
        result_.append(", subResult=" + getSubResult());
        result_.append(", subResultMixed=" + getSubResultMixed());
        result_.append("}");
        return result_.toString();
    }
}

File: dmn-test-cases/tck/1.1/cl3/0031-user-defined-functions/interpreter/expected/kotlin/dmn/type/TFnInvocationNamedResult.kt
package type

import java.util.*

@javax.annotation.Generated(value = ["itemDefinitionInterface.ftl", "tFnInvocationNamedResult"])
@com.fasterxml.jackson.annotation.JsonPropertyOrder(alphabetic = true)
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(`as` = type.TFnInvocationNamedResultImpl::class)
interface TFnInvocationNamedResult : com.gs.dmn.runtime.DMNType {
    @get:com.fasterxml.jackson.annotation.JsonGetter("subResult")
    val subResult: java.math.BigDecimal?

    @get:com.fasterxml.jackson.annotation.JsonGetter("subResultMixed")
    val subResultMixed: java.math.BigDecimal?

    @get:com.fasterxml.jackson.annotation.JsonGetter("divisionResultNamed")
    val divisionResultNamed: java.math.BigDecimal?

    @get:com.fasterxml.jackson.annotation.JsonGetter("multiplicationResultNamed")
    val multiplicationResultNamed: java.math.BigDecimal?

    override fun toContext(): com.gs.dmn.runtime.Context {
        val context = com.gs.dmn.runtime.Context()
        context.put("subResult", this.subResult)
        context.put("subResultMixed", this.subResultMixed)
        context.put("divisionResultNamed", this.divisionResultNamed)
        context.put("multiplicationResultNamed", this.multiplicationResultNamed)
        return context
    }

    fun equalTo(o: Any?): Boolean {
        if (this === o) return true
        if (javaClass != o?.javaClass) return false

        val other = o as TFnInvocationNamedResult
        if (if (this.divisionResultNamed != null) this.divisionResultNamed != other.divisionResultNamed else other.divisionResultNamed != null) return false
        if (if (this.multiplicationResultNamed != null) this.multiplicationResultNamed != other.multiplicationResultNamed else other.multiplicationResultNamed != null) return false
        if (if (this.subResult != null) this.subResult != other.subResult else other.subResult != null) return false
        if (if (this.subResultMixed != null) this.subResultMixed != other.subResultMixed else other.subResultMixed != null) return false

        return true
    }

    fun hash(): Int {
        var result = 0
        result = 31 * result + (if (this.divisionResultNamed != null) this.divisionResultNamed.hashCode() else 0)
        result = 31 * result + (if (this.multiplicationResultNamed != null) this.multiplicationResultNamed.hashCode() else 0)
        result = 31 * result + (if (this.subResult != null) this.subResult.hashCode() else 0)
        result = 31 * result + (if (this.subResultMixed != null) this.subResultMixed.hashCode() else 0)
        return result
    }

    fun asString(): String {
        val result_ = StringBuilder("{")
        result_.append("divisionResultNamed=" + divisionResultNamed)
        result_.append(", multiplicationResultNamed=" + multiplicationResultNamed)
        result_.append(", subResult=" + subResult)
        result_.append(", subResultMixed=" + subResultMixed)
        result_.append("}")
        return result_.toString()
    }

    companion object {
        @JvmStatic
        fun toTFnInvocationNamedResult(other: Any?): TFnInvocationNamedResult? {
            if (other == null) {
                return null
            } else if (other is TFnInvocationNamedResult?) {
                return other
            } else if (other is com.gs.dmn.runtime.Context) {
                var result_ = TFnInvocationNamedResultImpl()
                result_.subResult = other.get("subResult") as java.math.BigDecimal?
                result_.subResultMixed = other.get("subResultMixed") as java.math.BigDecimal?
                result_.divisionResultNamed = other.get("divisionResultNamed") as java.math.BigDecimal?
                result_.multiplicationResultNamed = other.get("multiplicationResultNamed") as java.math.BigDecimal?
                return result_
            } else if (other is com.gs.dmn.runtime.DMNType) {
                return toTFnInvocationNamedResult(other.toContext())
            } else {
                throw com.gs.dmn.runtime.DMNRuntimeException(String.format("Cannot convert '%s' to '%s'", other.javaClass.getSimpleName(), TFnInvocationNamedResult::class.java.getSimpleName()))
            }
        }
    }
}
